internal/versions/versions.go

```diff
@@ -103,6 +106,29 @@ func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params o
 			supportedArchs = []string{*releaseImage.CPUArchitecture}
 		}
 
+		// (MGMT-11859) We are filtering out multiarch release images so that they are available
+		//              only for customers allowed to use them. This is in order to be able to
+		//              expose them in OCP pre-4.13 without making them generally available.
+		if len(supportedArchs) > 1 {
+			hasMultiarchAuthorization := false
```

Comment:
I think we should use go-cache for this instead of using a couple of flags. It would allow expiration in case we want to grant/revoke perms, and should look nicer. For example see: https://github.com/openshift/assisted-service/blob/096a0f99b15dbb9a5afb71c86c146617511f14d6/pkg/auth/rhsso_authz_handler.go#L303